lib/looker-sdk/client/dynamic.rb

```diff
@@ -50,7 +50,20 @@ def load_swagger
         # We only need the swagger if we are going to be building our own 'operations' hash
         return if shared_swagger && @@sharable_operations[api_endpoint]
         # Try to load w/o authenticating. Else, authenticate and try again.
-        @swagger ||= without_authentication {try_load_swagger} || try_load_swagger
+        @swagger ||= without_authentication { try_load_swagger }
+        response_wo_auth = last_response
```

Comment:
this is odd because it assume that the right side of the ` @swagger ||= ...` executed. But, if `@swagger` was already non-nil then who knows what `last_response` would be set to?

Reply:
But, I suppose it doesn't matter because it is only used if you end up with `@swagger` being nil. I assume `last_response` is always safe to call.

Reply:
I'll double check that, thanks.

Reply:
Looks like `last_response` throws if called before any http requests have been made.  

I've adjusted the code to only touch last_response if @swagger was nil on the way in, so we did make an http request on the right hand side of the expression.